src/lgar.cxx

```diff
@@ -2432,6 +2537,17 @@ extern double lgar_theta_mass_balance(int layer_num, int soil_num, double psi_cm
     if (psi_cm_loc <= 0 && psi_cm_loc_prev < 1E-50) break;
 
     delta_mass_prev = delta_mass;
+
+
+    //so the above code that avoids infinite loops by breaking in certain cases works, but a fringe case must be addressed. 
+    //When theta is close to theta_r, the loop "while (delta_mass > tolerance)" will attempt to increase psi until mass balance closure occurs, but if the 
+    //AET value is large enough to make theta less than theta_r (or if dzdt was large enough to make the WF move such that the updated theta would be less than theta_r),
+    //the loop would never be able to achieve mass balance closure, and it will break when the one of above conditions is met.
```

Comment:
please remove the comments that are no longer relevant such as `theta less than theta_r`, we don't allow this condition anymore, correct?

Reply:
You're correct; we don't allow this condition. This comment is relevant however, because it explains that the following code addresses the case where theta would go below theta_r and instead prevents this.